src/commands/verify.ts

```diff
@@ -0,0 +1,20 @@
+import { CliVerifyArguments } from '../utils/define-args';
+import { generateReport } from '../utils/generate-report';
+import { getValidationResults } from '../utils/get-validation-results';
+
+export async function verify(argv: CliVerifyArguments) {
+    const validationResults = await getValidationResults(argv);
+    const hasAnyInteractionFailed = validationResults.some((result) =>
+        result.interactions.some(
+            (interaction) => interaction.validationResult === 'FAILED'
+        )
+    );
+
+    if (argv.outFile) {
+        await generateReport(argv.outFile, validationResults);
+    }
+
+    if (hasAnyInteractionFailed) {
+        process.exit(1);
```

Comment:
For consistency I think `process.exit` should be called `index.ts`. This method would only return the result